### PR TITLE
boards: arm: nrf: Adding updated Circuit Dojo board definitions.

### PR DESCRIPTION
**Pull request by jaredwolff**
Adding the Circuit Dojo nRF9160 Thing Plus device tree definitions.
Both secure and non secure targets.

Adding LIS2DH to nRF9160 Feather definitions.
Updating flash to W25Q32JV.

Signed-off-by: Jared Wolff <hello@jaredwolff.com>

### CODE REVIEW

**Comment by jaredwolff** [comment]
Need to update the pictures for the .rst file.

**Review by galak — DISMISSED** [review]
please remove the LICENSE file as Zephyr's main license is apache.

Also remove the README.md as docs are typically in rst for boards.

**Comment by jaredwolff** [comment]
On it. Those were left over from copying the contents from a separate repo.

**Comment by jaredwolff** [comment]
@galak addressed the two files.

File: boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_defconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_NRF91X=y
+CONFIG_SOC_NRF9160_SICA=y
+CONFIG_BOARD_SPARKFUN_THING_PLUS_NRF9160=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
```
Comment (ioannisg) [inline boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_defconfig:8]: Can you also enable HW_STACK_PROTECTION =y here? It's default now in cortex-m boards.

File: boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160ns_defconfig [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_NRF91X=y
+CONFIG_SOC_NRF9160_SICA=y
+CONFIG_BOARD_SPARKFUN_THING_PLUS_NRF9160NS=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
```
Comment (ioannisg) [inline boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160ns_defconfig:8]: ditto

Comment (jaredwolff) [reply boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160ns_defconfig:8]: Added this to the `ns` and regular variant.

File: boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_defconfig (hunk 1 above)
Comment (jaredwolff) [reply boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_defconfig:8]: Added this to the `ns` and regular variant.

**Comment by jaredwolff** [comment]
Not sure how to resolve the build issues or if it's necessary to commit these files. @ioannisg @carlescufi @galak anything you'd like to see to make this PR happen?

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**